sdk/core/azure-core/azure/core/rest/_rest.py

```diff
@@ -235,17 +235,23 @@ def encoding(self, value):
         # type: (str) -> None
         """Sets the response encoding"""
         self._encoding = value
+        self._text = None  # clear text cache
 
-    @property
-    def text(self):
-        # type: (...) -> str
-        """Returns the response body as a string"""
-        if self._text is None:
+    def text(self, encoding=None):
+        # type: (Optional[str]) -> str
+        """Returns the response body as a string
+
+        :param optional[str] encoding: The encoding you want to decode the text with. Can
+         also be set independently through our encoding property
+        :return: The response's content decoded as a string.
+        """
+        if self._text is None or encoding:
```

Comment:
It looks to me if text() is called once. Changing encoding will not take affect any more.

e.g. after called text(encoding="utf-8"), if I call text(encoding="ascii"), can I get the "ascii" decoded string?

Or is this by design?

Reply:
Never mind. :)